### PR TITLE
feat(localenv): span metrics generation

### PR DESCRIPTION
**Pull request by BlairCurrey**
## Changes proposed in this pull request

- Configures tempo to generate metrics based off spans
- Adds visualization for 95th percentile graphql resolver durations to localenv dashboard (live dashboard not applicable because tracing is local only atm)

Visualization preview:

image

## Context

fixes: https://github.com/interledger/rafiki/issues/2848

## Checklist

- [x] Related issues linked using `fixes #number`
- [ ] Tests added/updated
- [ ] Documentation added
- [x] Make sure that all checks pass
- [ ] Bruno collection updated

### CODE REVIEW

**Comment by BlairCurrey** [comment]
I considered if we wanted to add visualizations for each resolver. Like stat metrics for 25th, 50th, 95th percentile etc or something but opted not to.

One, I feel like we will better understand what details we need as we actually consume these (as part of performance testing analysis?). In addition, I think we probably mostly care about the extreme high end (ie 95th, 99th percentile etc). In which case maybe we just add another bar gauge like the included one but for 99th percentile.

**Comment by JoblersTune** [comment]
> live dashboard not applicable because tracing is local only atm

I'm curious exactly what our plan is with the local dashboard? Are we using it for dev? Are we using it to measure only certain local metrics. It's not exactly clear to me.

File: localenv/telemetry/grafana/provisioning/dashboards/example.json [hunk 1]
```diff
@@ -319,6 +320,76 @@
...
+          },
+          "editorMode": "code",
+          "expr": "histogram_quantile(0.95, sum(rate(traces_spanmetrics_latency_bucket{span_name=~\"^(mutation|query).*\"}[$__rate_interval])) by (le, span_name))",
+          "instant": false,
+          "legendFormat": "__auto",
+          "range": true,
+          "refId": "A"
+        }
+      ],
+      "title": "Panel Title",
```
Comment (mkurapov) [inline localenv/telemetry/grafana/provisioning/dashboards/example.json:390]: lets update this title

File: localenv/telemetry/grafana/provisioning/dashboards/example.json [hunk 2]
```diff
@@ -319,6 +320,76 @@
...
+      },
+      "pluginVersion": "11.1.3",
+      "targets": [
+        {
+          "datasource": {
+            "type": "prometheus",
+            "uid": "PBFA97CFB590B2093"
+          },
+          "editorMode": "code",
+          "expr": "histogram_quantile(0.95, sum(rate(traces_spanmetrics_latency_bucket{span_name=~\"^(mutation|query).*\"}[$__rate_interval])) by (le, span_name))",
```
Comment (mkurapov) [inline localenv/telemetry/grafana/provisioning/dashboards/example.json:383]: should this be something other than`$__rate_interval`, but instead the selected interval of the dashboard?

**Comment by mkurapov** [comment]
> I'm curious exactly what our plan is with the local dashboard? Are we using it for dev? Are we using it to measure only certain local metrics. It's not exactly clear to me.

Mainly for performance testing and debugging

**Comment by BlairCurrey** [comment]
> > live dashboard not applicable because tracing is local only atm
> 
> I'm curious exactly what our plan is with the local dashboard? Are we using it for dev? Are we using it to measure only certain local metrics. It's not exactly clear to me.

I'm mostly using it to validate the metric collection and develop visualizations. If it were applicable to the live version I would add them to there after merging this (although I guess technically it wouldn't have any data until the next release). I dont think we need tom maintain parity with the live version but its nice to have some basic proof-of-concept visualizations for some of the different metrics IMO.

Thinking back to our conversation about development workflow I think in theory it would be nice to develop locally, commit, then publishing to grafana from ci but not sure its worth the setup tbh..

File: localenv/telemetry/grafana/provisioning/dashboards/example.json (hunk 2 above)
Comment (BlairCurrey) [reply localenv/telemetry/grafana/provisioning/dashboards/example.json:383]: > should this be something other than$__rate_interval, but instead the selected interval of the dashboard?

From what I can tell it does factor in the current time range. I spun up the localenv, ran some queries and saw the data in this visualization with 5m time range. I waited 5m+ and saw no data until I bumped to 15m time range.

Im also seeing it generally recommended as starting point for the rate arg : https://grafana.com/blog/2020/09/28/new-in-grafana-7.2-__rate_interval-for-prometheus-rate-queries-that-just-work/

**Review by mkurapov — APPROVED** [review]

**Pull request merged**